examples/win-offsets.c

```diff
@@ -51,57 +78,164 @@ void sigint_handler()
     exit(1);
 }
 
+event_response_t cr3_cb(vmi_instance_t vmi, vmi_event_t *event)
+{
+    dp("CR#: %llx, VCPU: %d\n", event->reg_event.value, event->vcpu_id);
+
+    if (event->vcpu_id != 0) {
+        /* LibVMI initialization procedure strongly
+         * relies on reading CPU context for VCPU 0
+         * so we only do care about this one. */
```

Comment:
I guess this could also be adjusted. Nothing technically requires this to be vCPU 0, we just never bothered to update the code to perhaps check the other vCPUs or to pass it in as an input if we have a known vCPU with a good pagetable for the init to use. So I think going through all this trouble just to catch it on vCPU is a bit silly.

Reply:
Yes, VCPU=0 is hardcoded through whole initialization part. This could be easily changed in such way that user could hint which VCPU should be inspected.

Reply:
I've added `windows->init_vcpu` in the `windows_instance`. Using this value user may hint LibVMI to inspect the context of particular vCPU when doing `vmi_init_os()`. If not set, it's 0 by default, so it should be compatible with the old behavior of checking VCPU=0.